foodtruck/foodtruck.go

```diff
@@ -0,0 +1,108 @@
+/*
+Copyright © 2020 Chef Software, Inc <success@chef.io>
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package foodtruck
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/chef/foodtruck/connectors/azeventhub"
+)
+
+type Connector interface {
+	Register(string) error
+	StartListener() (chan []byte, error)
+	StopListener() error
+	Deregister() error
+	SendOrder(order []byte) error
+}
+
+type Provider interface {
```

Comment:
Is this for the future when other cloud providers are added?

Reply:
It's just providing an interface that any providers (like Chef/Inspec) need to adhere to.  For now, it will only be Chef/Inspec.